serow_ros/src/serow_driver.cpp

```diff
@@ -91,85 +89,67 @@ class SerowDriver {
     }
 
     void joint_state_topic_callback(const sensor_msgs::JointState::ConstPtr& msg) {
-        joint_state_data_.push(*msg);
-        if (joint_state_data_.size() > 100) {
-            joint_state_data_.pop();
-            ROS_WARN("SEROW is dropping joint state measurements, SEROW estimate is not real-time");
-        }
+        joint_state_data_ = *msg;
```

Comment:
Avoid using indicator flags like `gotJointMsg`, instead make this `joint_state_data_` an optional  and just clear it when used e.g. 

at line 109
```
if (joint_state_data_.has_value() && base_imu_data_.has_value()) {
  // Do work
  joint_state_data_.reset();
  base_imu_data_.reset();
  ros::spinOnce();
}